kube-core/src/resource.rs

```diff
@@ -44,15 +44,7 @@ pub trait Resource {
     /// Returns the plural name of the kind
     ///
     /// This is known as the resource in apimachinery, we rename it for disambiguation.
-    /// By default, we infer this name through pluralization.
-    ///
-    /// The pluralization process is not recommended to be relied upon, and is only used for
-    /// `k8s_openapi` types, where we maintain a list of special pluralisations for compatibility.
-    ///
-    /// Thus when used with `DynamicObject` or `kube-derive`, we override this with correct values.
-    fn plural(dt: &Self::DynamicType) -> Cow<'_, str> {
-        to_plural(&Self::kind(dt).to_ascii_lowercase()).into()
```

Comment:
Move `to_plural` function to `discovery.rs`?

Can we get rid of the special cases? (`kube_derive` can specify `plural` and `ApiResource::from_gvk_with_plural` exists)

Reply:
Ah, I was considering removing `ApiResource::from_gvk_with_plural` completely for something else. But was going to limit the scope of this release a bit.

I can make a follow-up PR though.